tempesta_fw/tls.c

```diff
@@ -577,10 +615,23 @@ tfw_tls_conn_drop(TfwConn *c)
 static int
 tfw_tls_conn_send(TfwConn *c, TfwMsg *msg)
 {
+	struct sk_buff *skb, *skb_head = NULL;
 	int r;
 	TlsCtx *tls = tfw_tls_context(c);
 	TlsIOCtx *io = &tls->io_out;
 
+	while ((skb = ss_skb_dequeue(&msg->skb_head))) {
+		if (skb_shinfo(skb)->tx_flags & SKBTX_SHARED_FRAG) {
```

Comment:
As far as I can tell, `SKBTX_DEV_ZEROCOPY` is set when MSG_ZEROCOPY is used:
```c
static inline struct ubuf_info *skb_zcopy(struct sk_buff *skb)
{
        bool is_zcopy = skb && skb_shinfo(skb)->tx_flags & SKBTX_DEV_ZEROCOPY;
        
        return is_zcopy ? skb_uarg(skb) : NULL;
}
```
```c
#define SKBTX_ZEROCOPY_FRAG     (SKBTX_DEV_ZEROCOPY | SKBTX_SHARED_FRAG)
```
So, maybe we should check for `SKBTX_ZEROCOPY_FRAG` rather than only for `SKBTX_SHARED_FRAG`?